pkg/replicas/getter_soc.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// the code below implements the integration of dispersed replicas in chunk fetching.
+// using storage.Getter interface.
+package replicas
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/ethersphere/bee/v2/pkg/file/redundancy"
+	"github.com/ethersphere/bee/v2/pkg/storage"
+	"github.com/ethersphere/bee/v2/pkg/swarm"
+)
+
+// getter is the private implementation of storage.Getter, an interface for
+// retrieving chunks. This getter embeds the original simple chunk getter and extends it
+// to a multiplexed variant that fetches chunks with replicas for SOC.
+//
+// the strategy to retrieve a chunk that has replicas can be configured with a few parameters:
+//   - SOCRetryInterval: the delay before a new batch of replicas is fetched.
+//   - depth: 2^{depth} is the total number of additional replicas that have been uploaded
+//     (by default, it is assumed to be 4, ie. total of 16)
+//   - (not implemented) pivot: replicas with address in the proximity of pivot will be tried first
+type socGetter struct {
+	wg sync.WaitGroup
```

Comment:
Correct me if I am wrong, but I do not see if WaitGroup is awaited?
Is it maybe required in Get:
```golang
    defer func() {
        g.wg.Wait()
    }()
```

Reply:
really nice catch, it is a problem in the cac replica getter as well.